src/cis/fam_cis_direct.cpp

```diff
@@ -1299,12 +1320,55 @@ Fam_Backup_Info Fam_CIS_Direct::get_backup_info(std::string BackupName,
     ostringstream message;
 
     CIS_DIRECT_PROFILE_START_OPS()
-    Fam_Memory_Service *memoryService = get_memory_service(memoryServerId);
+    auto obj = memoryServers->find(memoryServerId);
+    Fam_Memory_Service *memoryService = get_memory_service((uint64_t)obj->first);
     Fam_Backup_Info info = memoryService->get_backup_info(BackupName);
     CIS_DIRECT_PROFILE_END_OPS(cis_get_backup_info);
     return info;
 }
 
+std::string Fam_CIS_Direct::list_backup(std::string BackupName, uint64_t memoryServerId) {
+    ostringstream message;
+    string info;
+    auto obj = memoryServers->find(memoryServerId);
+    Fam_Memory_Service *memoryService = get_memory_service((uint64_t)obj->first);
+    info = memoryService->list_backup(BackupName);
+    return info;
+}
+
+void *Fam_CIS_Direct::delete_backup(string BackupName, uint64_t memoryServerId, uint32_t uid, uint32_t gid) {
+    ostringstream message;
+    auto obj = memoryServers->find(memoryServerId);
+    Fam_Memory_Service *memoryService = get_memory_service((uint64_t)obj->first);
+    Fam_Backup_Info info;
+    info = memoryService->get_backup_info(BackupName);
+    if (info.size == (int)-1) {
+        THROW_ERRNO_MSG(CIS_Exception,BACKUP_FILE_NOT_FOUND  ,"Backup doesnot exist.");
+    }
+    validate_backup(info, uid, gid,(S_IWOTH | S_IXOTH ));
```

Comment:
Validate backup for delete and List backup does not look secure enough.
Might have to re-think how to make it more secure.